include/libdnf/comps/environment/query.hpp

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright (C) 2021 Red Hat, Inc.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef LIBDNF_COMPS_ENVIRONMENT_QUERY_HPP
+#define LIBDNF_COMPS_ENVIRONMENT_QUERY_HPP
+
+#include "libdnf/common/sack/query.hpp"
+#include "libdnf/common/weak_ptr.hpp"
+#include "libdnf/comps/environment/environment.hpp"
+
+#include <memory>
+
+namespace libdnf {
+
+class Base;
+using BaseWeakPtr = WeakPtr<Base, false>;
```

Comment:
Please use the definition in `include/libdnf/base/base_weak.hpp` :slightly_smiling_face: